server/datastore/mysql/operating_system_vulnerabilities.go

```diff
@@ -203,51 +197,46 @@ func (ds *Datastore) DeleteOutOfDateOSVulnerabilities(ctx context.Context, src f
 	return nil
 }
 
-func (ds *Datastore) ListKernelsByOS(ctx context.Context, osID uint, teamID *uint) ([]*fleet.Kernel, error) {
+func (ds *Datastore) ListKernelsByOS(ctx context.Context, osVersionID uint, teamID *uint) ([]*fleet.Kernel, error) {
 	var kernels []*fleet.Kernel
+
 	stmt := `
 SELECT DISTINCT
 	software.id AS id,
 	software_cve.cve AS cve,
 	software.version AS version,
-    software_host_counts.hosts_count AS hosts_count
+    SUM(kernel_host_counts.hosts_count) AS hosts_count
```

Comment:
Needed this to handle the "All teams" case, where we don't care about team ID. Lmk if it makes sense.